Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,173 @@
+
+        foreach (var logMnemonic in logMnemonics.Where(x => x.index != mnemonicCurveIndex))
+        {
+            var logCurveMinMaxIndex = logCurveMinMaxIndexDictionary.GetValueOrDefault(logMnemonic.value, null);
+            if (logCurveMinMaxIndex == null)
+            {
+                continue;
+            }
+
+            var inputAnalyzeDataList = logDataRows
+                .Where(dataRow => !string.IsNullOrEmpty(dataRow[logMnemonic.index]))
+                .Select(dataRow => GetDepthOrDateTimeIndex(isDepthLog, dataRow[mnemonicCurveIndex], logCurveMinMaxIndex))
+                .Where(dataRowCurveIndex => ValidateCurveIndex(isLogIncreasing, logCurveMinMaxIndex)(dataRowCurveIndex))
+                .ToList();
+            
+            Index gapSize = isDepthLog && logCurveMinMaxIndex.MinIndex is DepthIndex depthMinIndex
+                ? new DepthIndex(job.GapSize, depthMinIndex.Uom)
+                : new TimeSpanIndex(job.TimeGapSize);
+
+            gapReportItems.AddRange(GetAnalyzeGapReportItem(logMnemonic.value, inputAnalyzeDataList, gapSize, isLogIncreasing));
+        }
+
+        return GetGapReportResult(job, gapReportItems, isDepthLog, logUid);
+    }
+
+    private static Func<Index, bool> ValidateCurveIndex(bool isLogIncreasing, LogCurveIndex logCurveIndex)
+    {
+        return isLogIncreasing
+            ? value => value >= logCurveIndex.MinIndex && value <= logCurveIndex.MaxIndex
+            : value => value <= logCurveIndex.MinIndex && value >= logCurveIndex.MaxIndex;
+    }
+
+    private static Index GetDepthOrDateTimeIndex(bool isDepthLog, string value, LogCurveIndex logCurveIndex)
+    {
+        return isDepthLog && logCurveIndex.MinIndex is DepthIndex depthMinIndex
+            ? new DepthIndex(double.Parse(value, CultureInfo.InvariantCulture), (depthMinIndex).Uom)
+            : new DateTimeIndex(DateTime.Parse(value, CultureInfo.InvariantCulture));
+    }
+
+    private (WorkerResult, RefreshAction) GetGapReportResult(AnalyzeGapJob job, List<AnalyzeGapReportItem> gapReportItems, bool isDepth, string logUid)
+    {
+        Logger.LogInformation("Analyzing gaps is done. {jobDescription}", job.Description());
+        job.JobInfo.Report = GetGapReport(gapReportItems, job.LogReference, isDepth);
+        WorkerResult workerResult = new(GetTargetWitsmlClientOrThrow().GetServerHostname(), true, $"Analyze gaps for log: {logUid}", jobId: job.JobInfo.Id);
+        return (workerResult, null);
+    }
+
+    private AnalyzeGapReport GetGapReport(List<AnalyzeGapReportItem> analyzeGapItems, LogObject logReference, bool isDepthLog)
+    {
+        return new AnalyzeGapReport
+        {
+            Title = $"Analyze gaps report",
+            Summary = analyzeGapItems.Count > 0
+                ? $"Found {analyzeGapItems.Count} gaps for {(isDepthLog ? "depth" : "time")} log '{logReference.Name}':"
+                : "No gaps were found for selected items.",
```

Comment:
Can we get a list of the mnemonics that has been analyzed? Otherwise you would have to remember which mnemonics the job has been run on. PR #2033 will include an accordion for the summary in the report modal when \n is included in the string to make more detailed summaries readable.

Reply:
fixed